finetune/lm_base.py

```diff
@@ -127,33 +152,43 @@ def _finetune(self, *Xs, Y, batch_size=None):
                 global_step += 1
                 if global_step % self.hparams.val_interval == 0:
 
-                    summary = self.sess.run(self.summaries, {self.X: xmb, self.M: mmb, self.Y: ymb})
+                    outputs = self._eval(
+                        self.summaries,
+                        feed_dict={
+                            self.X: xmb,
+                            self.M: mmb,
+                            self.Y: ymb
+                        }
+                    )
 
-                    self.train_writer.add_summary(summary, global_step)
+                    self.train_writer.add_summary(outputs.get(self.summaries), global_step)
 
                     sum_val_loss = 0
                     for xval, mval, yval in iter_data(*val_dataset, n_batch=n_batch_train, verbose=True):
-                        val_cost, summary = self.sess.run([self.clf_loss, self.summaries],
-                                                          {self.X: xval, self.M: mval, self.Y: yval,
-                                                           self.do_dropout: DROPOUT_OFF})
-                        self.valid_writer.add_summary(summary, global_step)
+                        outputs = self._eval(self.clf_loss, self.summaries, feed_dict={
+                            self.X: xval, self.M: mval, self.Y: yval,
+                            self.do_dropout: DROPOUT_OFF
+                        })
+                        self.valid_writer.add_summary(outputs.get(self.summaries), global_step)
+
+                        val_cost = outputs.get(self.clf_loss, 0)
                         sum_val_loss += val_cost
-                        avg_val_loss = avg_val_loss * self.hparams.rolling_avg_decay + val_cost * (
-                                1 - self.hparams.rolling_avg_decay)
-                        _LOGGER.info("\nVAL: LOSS = {}, ROLLING AVG = {}".format(val_cost, avg_val_loss))
+                    
                     val_window.append(sum_val_loss)
                     val_window.pop(0)
 
                     if np.mean(val_window) <= best_val_loss:
                         best_val_loss = np.mean(val_window)
-                        _LOGGER.info("Autosaving new best model.")
+                        _LOGGER.info("Autosaving new best model...")
                         self.save(self.autosave_path)
-                        _LOGGER.info("Done!!")
-                cost, _ = self.sess.run([self.clf_loss, self.train_op],
-                                        {self.X: xmb, self.M: mmb, self.Y: ymb, self.do_dropout: DROPOUT_ON})
-                avg_train_loss = avg_train_loss * self.hparams.rolling_avg_decay + cost * (
-                        1 - self.hparams.rolling_avg_decay)
-                _LOGGER.info("\nTRAIN: LOSS = {}, ROLLING AVG = {}".format(cost, avg_train_loss))
```

Comment:
I prefer to encourage use of tensorboard here rather than outputting logs -- if for no other reason than it makes the tqdm progress logging look atrocious.